agent/agent.go

```diff
@@ -2042,25 +2144,13 @@ func (a *Agent) AddCheck(check *structs.HealthCheck, chkType *structs.CheckType,
 	}
 
 	if check.ServiceID != "" {
-		s := a.State.Service(check.ServiceID)
-		if s == nil {
-			return fmt.Errorf("ServiceID %q does not exist", check.ServiceID)
-		}
-		check.ServiceName = s.Service
-		check.ServiceTags = s.Tags
+		check.ServiceName = service.Service
+		check.ServiceTags = service.Tags
```

Comment:
Can't `service` be nil in some cases? What if this is a check registration with a service ID that doesn't exist? Even if we can't have that because of validation nil panics scare me - can we nil check here and error maybe?

Reply:
Shound't be nil but added a check to be extra safe.